### PR TITLE
fix: Fetch ref in pipelinerunner version stream checkout

### LINKED ISSUE #5195: clientFactory.cloneVersionStream does not work for shas
If the version stream ref is a SHA, the meta pipeline clientfactory cannot clone the versions repo:

```
kubectl logs pipelinerunner-6ccccd5454-8g68f
error: unable to clone version dir: unable to clone version stream: Cloning into '.'...
warning: Could not find remote branch f86ee6eb98ed9769c9f628ce91242bb954296978 to clone.
fatal: Remote branch f86ee6eb98ed9769c9f628ce91242bb954296978 not found in upstream origin: failed to run 'git clone --depth 1 --branch f86ee6eb98ed9769c9f628ce91242bb954296978 https://github.com/jenkins-x/jenkins-x-versions.git .' command in directory '/tmp/jx-version-repo-730349822', output: 'Cloning into '.'...
warning: Could not find remote branch f86ee6eb98ed9769c9f628ce91242bb954296978 to clone.
fatal: Remote branch f86ee6eb98ed9769c9f628ce91242bb954296978 not found in upstream origin'
```

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

If we don't do this, we can't actually checkout the ref.

#### Special notes for the reviewer(s)

/assign @hferentschik 
/assign @rawlingsj 
/assign @pmuir 

#### Which issue this PR fixes

fixes #5195

### CODE REVIEW

**Comment by abayer** [comment]
Worth mentioning that I tried this exact set of commands out by hand and it worked.

File: pkg/tekton/metapipeline/clientfactory.go [hunk 1]
```diff
@@ -266,6 +266,18 @@ func cloneVersionStream(url string, ref string) (string, error) {
 			return "", errors.Wrapf(err, "unable to clone version stream: %s", output)
 		}
 
+		// Fetch PR refs before checking out the ref
```
Comment (hferentschik) [inline pkg/tekton/metapipeline/clientfactory.go]: 👍

**Pull request merged**